lib/intercom/index.js

```diff
@@ -86,29 +86,31 @@ Intercom.prototype.identify = function(identify){
 
   traits.app_id = this.options.appId;
 
-  // intercom requires `company` to be an object. default it with group traits
-  // so that we guarantee an `id` is there, since they require it
-  if (null != traits.company && !is.object(traits.company)) delete traits.company;
-  if (traits.company) defaults(traits.company, group.traits());
-
   // name
   if (name) traits.name = name;
 
   // handle dates
-  if (created) {
+  if (identify.created()) {
     del(traits, 'created');
     del(traits, 'createdAt');
-    traits.created_at = created;
+    traits.signed_up_at = identify.created();
   }
-  if (companyCreated) {
+  if (identify.companyCreated()) {
     del(traits.company, 'created');
     del(traits.company, 'createdAt');
-    traits.company.created_at = companyCreated;
+    traits.company.created_at = identify.companyCreated();
   }
 
   // convert dates
   traits = convertDates(traits, formatDate);
 
+  var companies = dot(traits, 'companies');
+  var company = dot(traits, 'company');
```

Comment:
Are you expecting `traits` to potentially be undefined? If not, dot notation is clearer here